classes/DwcArchiverOccurrence.php

```diff
@@ -414,223 +414,236 @@ public function getExsiccateArr($occid){
-			if($rs){
-				while($r = $rs->fetch_object()){
-					$assocKey = 'sd-'.$r->occidAssociate;
-					$assocArr[$assocKey]['occidassoc'] = $r->occidAssociate;
-					$assocArr[$assocKey]['relationship'] = 'herbariumSpecimenDuplicate';
-					$relOccidArr[$r->occidAssociate][] = $assocKey;
-				}
-				$rs->free();
-			}
+			//Append associations of duplicate specimen
+			$this->appendSpecimenDuplicateAssociations($occid, $assocArr, $internalAssocOccidArr);
+
 			//Append resource URLs to each output record
-			if($relOccidArr){
-				$this->setServerDomain();
-				//Replace GUID identifiers with occurrenceID values
-				$sql = 'SELECT occid, occurrenceID, recordID FROM omoccurrences WHERE occid IN('.implode(',',array_keys($relOccidArr)).')';
-				$rs = $this->conn->query($sql);
-				while($r = $rs->fetch_object()){
-					foreach($relOccidArr[$r->occid] as $k => $targetAssocID){
-						if($r->occurrenceID){
-							$assocArr[$targetAssocID]['resourceurl'] = $r->occurrenceID;
-							if(substr($r->occurrenceID, 0, 4) != 'http'){
-								$assocArr[$targetAssocID]['resourceurl'] = $this->serverDomain.$GLOBALS['CLIENT_ROOT'].'/collections/individual/index.php?guid='.$r->occurrenceID;
-							}
-						}
-						elseif($r->recordID){
-							$assocArr[$targetAssocID]['resourceurl'] = $this->serverDomain.$GLOBALS['CLIENT_ROOT'].'/collections/individual/index.php?guid='.$r->recordID;
-						}
+			if($internalAssocOccidArr){
+				$identifierArr = $this->getInternalResourceIdentifiers($internalAssocOccidArr);
+				foreach($identifierArr as $internalOccid => $idArr){
+					foreach($internalAssocOccidArr[$internalOccid] as $targetAssocID){
+						$assocArr[$targetAssocID] = array_merge($assocArr[$targetAssocID], $idArr);
 					}
 				}
-				$rs->free();
 			}
 			//Create output strings
 			$retStr = '';
 			foreach($assocArr as $assocateArr){
-				$retStr .= ' | '.$assocateArr['relationship'];
-				if(!empty($assocateArr['subtype'])) $retStr .= ' ('.$assocateArr['subtype'].')';
-				$retStr .= ': '.$assocateArr['resourceurl'];
+				if($associationType == 'observational'){
+					$retStr .= ' | '.$assocateArr['relationship'];
+					if(!empty($assocateArr['scientificName'])) $retStr .= ': '.$assocateArr['scientificName'];
+				}
+				else{
+					$retStr .= ' | relationship: '.$assocateArr['relationship'];
+					if(!empty($assocateArr['subtype'])) $retStr .= ' ('.$assocateArr['subtype'].')';
+					if(!empty($assocateArr['identifier'])) $retStr .= ', identifier: '.$assocateArr['identifier'];
+					if(!empty($assocateArr['basisOfRecord'])) $retStr .= ', basisOfRecord: '.$assocateArr['basisOfRecord'];
+					if(!empty($assocateArr['scientificName'])) $retStr .= ', scientificName: '.$assocateArr['scientificName'];
+					if(!empty($assocateArr['resourceUrl'])) $retStr .= ', resourceUrl:  '.$assocateArr['resourceUrl'];
+					if(empty($assocateArr['identifier']) && !empty($assocateArr['recordID'])) $retStr .= ', recordID:  '.$assocateArr['recordID'];
```

Comment:
should `empty($assocateArr['identifier'])` be `!empty($assocateArr['identifier'])` here?

I'm not following the logic, but the pattern looks different.

Reply:
@egbot so, this is not a typo?